src/plugins/hiddenColumns/test/contextMenu/hideColumn.e2e.js

```diff
@@ -0,0 +1,63 @@
+describe('ContextMenu', () => {
+  const id = 'testContainer';
+
+  beforeEach(function() {
+    this.$container = $(`<div id="${id}"></div>`).appendTo('body');
+  });
+
+  afterEach(function() {
+    if (this.$container) {
+      destroy();
+      this.$container.remove();
+    }
+  });
+
+  describe('hide column', () => {
+    it('should hide the entry for "Hide column" in the context menu, when the selection doesn\'t contain an' +
+      ' entire column (including the header)', () => {
+      handsontable({
+        data: Handsontable.helper.createSpreadsheetData(4, 4),
+        contextMenu: true,
+        rowHeaders: true,
+        colHeaders: true,
+        hiddenColumns: true,
+      });
+
+      selectCell(0, 0);
+      contextMenu();
+
+      const contextMenuPlugin = getPlugin('contextMenu');
+      const hiddenColumnsCMEntry = contextMenuPlugin.menu.menuItems.filter((el) => {
```

Comment:
I think that it is not the best idea for messing with internal properties in the _ContextMenu_ menu instance. It's bad for future refactoring. Can I ask you to use a more "end" approach? Something like grabbing available options from the context menu by generating DOM texts `$('.htContextMenu tbody td').text()` or something similar.

Reply:
That's basing on strings, which might change at any point, and that's not great either.

Reply:
When the string changed this can be helpful for catching that kind of anomalies. If you have an idea of how to improve the tests using the "end" approach I'm open for suggestions.

Reply:
If we change the string from "Hide column" to "Hide column here" (or something like this), it can be visible, and this kind of test would give us a false positive result (it would search for "Hide column", not find it, and assume it's not there).

I can change it, if you insist, but I don't think it's more bulletproof than my approach.

Reply:
Yep, it's hard to find the best possible solution here. I'd use regex or `includes` to catch the "Hide column" phrase.